lighthouse-core/lib/url-shim.js

```diff
@@ -96,6 +96,19 @@ URL.getURLDisplayName = function getURLDisplayName(url, options) {
   return Util.getURLDisplayName(new URL(url), options);
 };
 
+/**
+ * @param {string} url
+ * @return {string}
+ */
+URL.elideDataURI = function elideDataURI(url) {
+  try {
+    const parsed = new URL(url);
+    return parsed.protocol === 'data:' ? url.slice(0, 100) : url;
```

Comment:
yeah, I guess I'm thinking about `URL.elideDataURI` in isolation/other future uses. To me it makes sense to get a shortened version of the canonicalized URL out of calling it, but without a known change that could be made to it, up to you